src/Gateway.php

```diff
@@ -201,8 +203,12 @@ private function setBaseUrl()
     {
         if (self::API_VERSION_EUROPE === $this->getApiRegion()) {
             $this->parameters->set('base_url', $this->getTestMode() ? self::EU_TEST_BASE_URL : self::EU_BASE_URL);
-        } else {
-            $this->parameters->set('base_url', $this->getTestMode() ? self::NA_TEST_BASE_URL : self::NA_BASE_URL);
+
+            return $this;
```

Comment:
No need to return self in a private function, this can just be `return` (and the final return can be dropped altogether)